cse/README.md

```diff
@@ -1,31 +1,83 @@
+  * [GCP cluster creation](https://cloud.google.com/kubernetes-engine/docs/how-to/creating-a-zonal-cluster)
+  * [Azure cluster creation](https://docs.microsoft.com/en-us/azure/aks/kubernetes-walkthrough-portal)
+* [Helm is installed](https://helm.sh/docs/intro/install/) on your terminal.
+* Your terminal is connected to your Kubernetes cluster and ready to use `kubectl`
 
-```
-{ "virtru-test": "http://jwt.default.svc.cluster.local/jwk.json" }
-```
+## Installation Steps
+### Configure IDP
+
+To use Google's CSE service, you must have a 3rd party identity provider configured to authenticate users to the CSE service. Documentation on Google's requirements can be found [here](https://support.google.com/a/answer/10743588?hl=en).
+
+### Provision SSL Certificate
+
+Virtru's KMS for Google CSE runs on a secure connection from Google to the service. The certificates, for this service, will be mounted into the running container. When filling out the `values.yaml` file in the section below, you will need the private key and certificate chain available to you.
 
-`jwtKaclsUrl` should have the format
+### Updating `values.yaml` file
 
+This section will detail potential changes that you will need to make to your `values.yaml` file.
+
+#### `appConfig`
+
+* `jwksAuthnIssuers` - A base-64 encoded map of accepted Authentication issuer ids (from the authentication JWT) to the URL where the issuer publishes its JSON Web Keyset. This is dictated via the IDP. In the example provided it is Google OAuth
+  * Example command to base-64 encode: 
+    * ```
+        echo '{ "https://accounts.google.com": "https://www.googleapis.com/oauth2/v3/certs" }' | base64
+        ``` 
+* `jwksAuthzIssuers` - A base-64 encoded map of accepted Authorization issuer ids (from the authorization JWT) to the URL where the issuer publishes its JSON Web Keyset. This is dictated by Google and is filled out by default
+* `jwtAud` - A base-64 encoded JSON map of JWT audiences for authorization and authentication. The `authz` audience, which is sent by Google, will always be  `cse-authorization`, but the `authn` audience will be configured through the customer’s IDP. In the example provided the `authn` audience is Google OAuth
+  * Example command to base-64 encode: 
+    * ```
+        echo '{ "authn": "00000000000000000.apps.googleusercontent.com", "authz":"cse-authorization" }' | base64
+        ```
+* `jwtKaclsUrl` - URL for your CSE service. This should match the SSL certificate provisioned in the previous steps
+* `useCks` - Default false. Switch to true if using a Virtru CKS in tandem with your CSE KMS
+*  `cksUrl` - Leave as default if not using CKS. If using CKS, this is the FQDN of your running CKS service (example: `https://cks.example.com`)
+
+#### `appSecrets` 
+
+In the `appSecrets` section, the `hmac`, `secretKey`, and `cksHmac` (if using CKS) sections must be the plaintext values for your secrets, while in `ssl` you must base-64 encode the private key and certificate.
```

Comment:
are we doing values.yaml instead of files here in anticipation of getting this into marketplace? (if so -- might need a new solution to pull in this chart in the same way as the others so probably not really necessary to plan around that here)

not going to block on it either way as I know we've talked about secrets ad-nauseum at this point and you got the go-ahead from sean/ben too, just pointing out that we don't need values.yaml here as this won't touch marketplace

Reply:
Wait so this chart for CSE isn't going to touch marketplace at all? 🤔  Happy change it to external secret creation if so, I just thought we were keeping CSE as is because it pushes to the marketplace deployment

Reply:
Oh sorry my bad -- for some reason I thought I was looking at gateway not cse